rust-crates/symblib/src/symbconv/go.rs

```diff
@@ -67,18 +67,45 @@ fn extract_ranges(obj: &objfile::Reader<'_>, visitor: super::RangeVisitor<'_>) -
     stats.is_go_binary = true;
 
     let mut func_iter = go.funcs()?;
-    while let Some(func) = func_iter.next()? {
+    'outer: while let Some(func) = func_iter.next()? {
         // Infer end of function from line tables.
-        let Some(end) = func.line_mapping()?.map(|(rng, _)| Ok(rng.end)).max()? else {
-            debug!(
-                "WARN: unable to determine end of function ({})",
-                func.name()?
-            );
-            stats.funcs_skipped += 1;
-            continue;
+        let mut end: Option<u64> = None;
+        let mut line_iter = func.line_mapping()?;
+        loop {
+            match line_iter.next() {
+                Ok(Some((rng, _))) => {
+                    if let Some(e) = end {
+                        if rng.end < e {
+                            continue;
+                        }
+                    }
+                    end = Some(rng.end)
+                }
+                Ok(None) => break,
+                Err(gosym::Error::BadLineNumber) => {
+                    // skip function once we hit a BadLineNumber error
+                    stats.funcs_skipped += 1;
+                    debug!("WARN: bad line number of function ({})", func.name()?);
+                    continue 'outer;
+                }
+                Err(e) => {
+                    return Err(e.into());
```

Comment:
Just a thought and maybe we can add more context to the error.
```suggestion
                    return Err(Error::GoSymbol(e).context("Error in line mapping"));
```

Reply:
Still fairly new to rust, but do think this would require me to use [anyhow](https://docs.rs/anyhow/latest/anyhow/trait.Context.html)?

I have defined a new Error type `GoSymbolBadLineMapping`, that transports this context. Let me know you think.